middleware/redirectNoQueue.js

```diff
@@ -0,0 +1,37 @@
+const { baseUrl } = require('../util')
+const { Queue } = require('../models')
+
+module.exports = async (req, res, next) => {
+  const { queueId: qid } = req.params
+  const firstQueue = await Queue.findOne({
+    where: {
+      id: qid,
+    },
+    paranoid: false,
+  })
+
+  // Cannot Find Queue In Deleted or Non Deleted
+  if (!firstQueue) {
+    res.redirect(`${baseUrl}/`)
```

Comment:
You'll want to return after this so you don't try to access any properties of `null` in the next if statement. Alternatively, make this an if-else-if with the next block.